src/lib.rs

```diff
@@ -120,12 +120,39 @@ impl<Kind, Type> IdMap<Kind, Type> where Type: Eq + Clone, Kind: Clone {
     }
 }
 
-type SyncExtSender = Mutex<Box<ExtSender<WatchEvent>>>;
-type WatchersMap = HashMap<usize, Arc<SyncExtSender>>;
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum SendDirection {
+    Enter,
+    Exit,
+}
+
+trait RangeChecker {
```

Comment:
1/ Could you document this in a followup?
2/ You actually don't need a trait for that, right? I think that you just want to
```rust
impl Option<Range> {
  fn should_send(...) {
     // ...
  }
}
```